### DISCUSSION TITLE
Can TLS-Fuzzer Simulate Handshake with Specific PSK for Encrypting Data?

### DISCUSSION THREAD

**Discussion by hamma96** [tlsfuzzer/tlsfuzzer#961] [Q&A]
**Discussion:**

I am using **TLS-Fuzzer** to simulate a full TLS handshake with a pre-shared key (PSK) and would like to know if it's possible to configure a specific PSK value for the encrypted handshake messages.

### Scenario:

- **Client:** TLS-Fuzzer
- **Server:** A dummy web server (acting as the TLS server)
  
In my specific use case, the key for encrypting the third message in the TLS 1.2 handshake (i.e., the **Client Finished** message) should be derived from the following parameters:

1. Client random number
2. Server random number
3. Pre-Shared Key (PSK)

### Question:

How can I set a specific PSK value in **TLS-Fuzzer** to be used during the handshake process, particularly for generating the encryption key for the Client Finished message?

I have not been able to find documentation or examples on how to configure the PSK in the TLS-Fuzzer. Is there an existing way to:

1. **Set a custom PSK** for the handshake?
2. Ensure the PSK is used in conjunction with the Client and Server random values to generate the key for encryption?

Any guidance or examples would be greatly appreciated!

**Comment by tomato42** [comment]
There's no support for PSK in TLS 1.2 because tlslite-ng lacks this support: https://github.com/tlsfuzzer/tlslite-ng/issues/22 

There's [test-tls13-psk_ke.py](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/scripts/test-tls13-psk_ke.py) and [test-tls13-psk_dhe_ke.py](https://github.com/tlsfuzzer/tlsfuzzer/blob/master/scripts/test-tls13-psk_dhe_ke.py) but as the name suggests, they're limited to TLS 1.3
